src/System.Management.Automation/engine/Modules/GetModuleCommand.cs

```diff
@@ -341,6 +351,22 @@ protected override void ProcessRecord()
                 ThrowTerminatingError(error);
             }
 
+            // SkipCompatibilityCheck should not work with PSEdition specified
```

Comment:
It seems that the two are not mutually exclusive.  `-PSEdition` is for filtering while `-SkipCompatibilityCheck` is for enforcement.